cocotb/triggers.py

```diff
@@ -751,7 +728,7 @@ def on_done(ret):
             waiters.append(cocotb.fork(_wait_callback(t, on_done)))
 
         # wait for a waiter to complete
-        yield e.wait()
+        await e.wait()
 
         # kill all the other waiters
         # TODO: Should this kill the coroutines behind any Join triggers?
```

Comment:
Has this behavior changed?

Reply:
Do you mean the TODO line, the "kill all the waiters" line, or the change?